lms/product/blackboard/_plugin/grouping.py

```diff
@@ -19,12 +20,13 @@ class BlackboardGroupingPlugin(GroupingPlugin):
     group_type = Grouping.Type.BLACKBOARD_GROUP
     sections_type = None  # We don't support sections in Blackboard
 
-    def __init__(self, blackboard_api):
+    def __init__(self, blackboard_api, group_set_service: GroupSetService):
         self._blackboard_api = blackboard_api
+        self._group_set_service = group_set_service
 
     def get_group_sets(self, course: Course):
         group_sets = self._blackboard_api.course_group_sets(course.lms_id)
-        course.set_group_sets(group_sets)
+        self._group_set_service.store_group_sets(course, group_sets)
```

Comment:
There's going to bunch like this, just replace the call to the model's method with the new service one.